app/workers/billing_worker.rb

```diff
@@ -45,6 +45,13 @@ def self.enqueue(provider, billing_date, buyers_scope = nil)
   end
 
   def self.enqueue_for_buyer(buyer, billing_date)
+    if buyer.id == buyer.provider_account_id
+      System::ErrorReporting.report_error(ArgumentError.new("invalid buyer #{buyer.id}, has self as provider"),
```

Comment:
Why not `SpuriousBillingError` here?

Reply:
My line of thinking was that for `::enqueue_for_buyer` method it is an invalid argument to supply not a buyer. Also Also will differentiate better where we get the error from if that matters.